tests/test_screen_interpolator.py

```diff
@@ -31,33 +37,137 @@ def test_nearest_neighbor_interpolation(duration, fps, image_frame_count, num_vi
 
         root = Path("tests/screen_data")
         data_dir = root / "data"
+        meta_dir = root / "meta"
 
-        npy_files = [
-            f
-            for f in sorted(root.glob("*.npy")) + sorted(data_dir.glob("*.npy"))
-            if f.name != "timestamps.npy"
-        ]
-
-        frames = []
-
-        for fpath in npy_files:
-            arr = np.load(fpath)
-            if arr.ndim == 3:
-                for i in range(arr.shape[0]):
-                    frames.append(arr[i])
-            elif arr.ndim == 2:
-                frames.append(arr)
-            else:
-                raise ValueError(f"Unexpected array shape {arr.shape} in {fpath}")
+        # Load frames based on encoding format
+        frames = _load_frames_from_data_dir(data_dir, meta_dir, encoded)
 
         raw_array = np.stack(frames)
-
         expected_indices = np.round((times - timestamps[0]) * fps).astype(int)
         expected_frames = raw_array[expected_indices]
 
-        interp, valid = interp_obj.interpolate(times=times)
-
-        assert times.shape == valid.shape, "All interpolated frames should be valid"
-        assert np.allclose(
-            interp, expected_frames, atol=1e-5
-        ), "Nearest neighbor interpolation mismatch"
+        interp = interp_obj.interpolate(times=times)
+
+        if expected_frames.max() > 1.1:
```

Comment:
why do we check if its more than 1.1 and not just strictly above 1 ?
the idea is the check if the values are between 0 and 1 or between 0 and 255, right?